src/test/java/se/iths/springbootgroupproject/controllers/WebControllerTest.java

```diff
@@ -0,0 +1,247 @@
+                "First message",
+                "FirstUser");
+        message2 = new MessageAndUsername(2L,
+                LocalDate.now(),
+                LocalDate.now(),
+                "Second",
+                "Second message",
+                "SecondUser");
+    }
+
+    @Test
+    @WithAnonymousUser
+    void redirectUnauthenticatedUsersToLogin() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/web/messages"))
+                .andExpectAll(
+                        status().is3xxRedirection(),
+                        redirectedUrl("http://localhost/oauth2/authorization/github"));
+    }
+
+    @Test
+    @WithMockUser
+    void shouldShowAllMessagesToAuthenticated() throws Exception {
+        List<MessageAndUsername> messages = List.of(message, message2);
+        when(messageService.findAllMessages(any(Pageable.class))).thenReturn(messages);
+        mockMvc.perform(MockMvcRequestBuilders.get("/web/messages"))
+                .andExpectAll(
+                        status().isOk(),
+                        model().attribute("messages", messages),
+                        view().name("messages"));
+    }
+
+    @Test
+    @WithAnonymousUser
+    void shouldShowPublicMessagesToUnauthenticatedUser() throws Exception {
+        List<MessageAndUsername> messages = List.of(message, message2);
+        when(messageService.findAllByPrivateMessageIsFalse(any(Pageable.class))).thenReturn(messages);
+        mockMvc.perform(MockMvcRequestBuilders.get("/web/welcome"))
+                .andExpectAll(status().isOk(),
+                        model().attribute("messages", messages),
+                        view().name("welcome"));
+    }
+
+    @Test
+    void authenticatedUserAccessTheirProfile() throws Exception {
+        User user = new User();
+        user.setFirstName("Pelle");
+        user.setEmail("Pelle@gmail.com");
+        user.setUserName("Pelle2k");
+        user.setLastName("Pellsson");
+        user.setImage("/img.jpg");
+        when(userService.findByGitHubId(any())).thenReturn(user);
+        mockMvc.perform(MockMvcRequestBuilders.get("/web/myprofile")
+                        .with(oauth2Login()))
+                .andExpectAll(
+                        model().attribute("name", user.getFirstName()+" "+user.getLastName()),
+                        model().attribute("email", user.getEmail()),
+                        model().attribute("userName", user.getUserName()),
+                        model().attribute("profilepic", user.getImage()),
+                        status().isOk(),
+                        view().name("userprofile"));
+    }
+
+    @Test
+    @WithAnonymousUser
+    void redirectedIfNotAuthenticated() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/web/myprofile/edit"))
+                .andExpectAll(
+                        status().is3xxRedirection(),
+                        redirectedUrl("http://localhost/oauth2/authorization/github"));
+    }
```

Comment:
You have made two tests for testing redirection if not authenticated. Is there a reason for there to be two?

Reply:
First one is on row 68-75

Reply:
No real reason just two random endpoints tested for redirection to make sure it works. Best practice would probably be to cover every redirect case in controller..

Reply:
Okay, thought you maybe got lost in the code and added a second test :) I see why you have it there now, good thinkin'! 👌